examples/log/boxfddp_vs_boxddp.log

```diff
@@ -17,74 +17,74 @@ iter 	 cost 	      stop 	    grad 	  xreg 	      ureg 	 step 	 feas
-  16  4.07553e+06  1.75344e+09  1.16690e+07  1.00000e-01  1.00000e-01   0.0625     1
-  17  3.95576e+06  2.15229e+09  9.39190e+06  1.00000e+00  1.00000e+00   0.0078     1
-  18  2.57884e+06  1.94745e+09  8.59791e+06  1.00000e+00  1.00000e+00   0.2500     1
-  19  1.98165e+06  4.96524e+08  4.68312e+06  1.00000e+00  1.00000e+00   0.1250     1
+  10  7.94788e+06  1.27689e+10  2.16983e+07  1.00000e-03  1.00000e-03   0.0078     1
+  11  7.84938e+06  1.20599e+10  2.09092e+07  1.00000e-02  1.00000e-02   0.0039     1
+  12  7.70102e+06  1.27440e+10  2.35580e+07  1.00000e-01  1.00000e-01   0.0039     1
+  13  7.02848e+06  1.91080e+10  1.86348e+07  1.00000e-01  1.00000e-01   0.0625     1
+  14  6.52271e+06  2.63294e+10  2.16867e+07  1.00000e-01  1.00000e-01   0.0156     1
+  15  5.59608e+06  4.54927e+10  1.98268e+07  1.00000e-01  1.00000e-01   0.0312     1
+  16  5.38821e+06  1.44638e+10  1.63123e+07  1.00000e+00  1.00000e+00   0.0078     1
+  17  4.97903e+06  9.86993e+09  1.58666e+07  1.00000e+00  1.00000e+00   0.1250     1
+  18  4.07537e+06  6.07056e+09  1.36559e+07  1.00000e+00  1.00000e+00   0.0312     1
+  19  2.70770e+06  6.19772e+09  1.02574e+07  1.00000e+00  1.00000e+00   0.0625     1
 iter 	 cost 	      stop 	    grad 	  xreg 	      ureg 	 step 	 feas
-  20  1.59685e+06  3.79602e+08  3.03430e+06  1.00000e+00  1.00000e+00   0.2500     1
-  21  1.22252e+06  2.16068e+09  2.66424e+06  1.00000e+00  1.00000e+00   0.2500     1
-  22  7.50969e+05  1.26130e+09  1.49246e+06  1.00000e+00  1.00000e+00   0.5000     1
-  23  6.11939e+05  3.28316e+08  8.24457e+05  1.00000e+00  1.00000e+00   0.5000     1
-  24  5.06824e+05  2.06988e+08  8.37232e+05  1.00000e+00  1.00000e+00   0.2500     1
-  25  3.98574e+05  4.92668e+07  4.99563e+05  1.00000e+00  1.00000e+00   0.5000     1
-  26  3.00385e+05  8.45030e+07  2.71597e+05  1.00000e+00  1.00000e+00   0.5000     1
-  27  2.80569e+05  1.91150e+07  1.17780e+05  1.00000e+00  1.00000e+00   0.5000     1
-  28  2.46594e+05  1.42174e+07  1.10356e+05  1.00000e+00  1.00000e+00   0.5000     1
-  29  2.30612e+05  6.38317e+06  4.71698e+04  1.00000e+00  1.00000e+00   0.5000     1
+  20  2.17520e+06  1.24754e+09  6.09848e+06  1.00000e+00  1.00000e+00   0.0625     1
+  21  1.79919e+06  5.33754e+08  3.76591e+06  1.00000e+00  1.00000e+00   0.1250     1
+  22  1.13695e+06  9.05971e+08  2.59785e+06  1.00000e+00  1.00000e+00   0.5000     1
+  23  8.30290e+05  1.41761e+08  1.93369e+06  1.00000e+00  1.00000e+00   0.5000     1
+  24  5.77495e+05  6.77960e+07  1.13559e+06  1.00000e+00  1.00000e+00   0.2500     1
+  25  4.27336e+05  2.77789e+07  4.46809e+05  1.00000e+00  1.00000e+00   0.5000     1
+  26  3.91060e+05  2.39396e+07  2.19975e+05  1.00000e+00  1.00000e+00   0.2500     1
+  27  3.72508e+05  1.91335e+07  1.63448e+05  1.00000e+00  1.00000e+00   0.1250     1
+  28  3.45084e+05  1.43382e+07  1.30711e+05  1.00000e+00  1.00000e+00   0.5000     1
+  29  3.27129e+05  1.71809e+07  1.18675e+05  1.00000e+00  1.00000e+00   0.5000     1
```

Comment:
The other log file changes are minor. This is quite significant. Any insights into why it disimproved this much? I.e. the accepted step lengths are quite a few steps smaller than before

Reply:
I forgot to mention this, thank you for bringing the topic.

There are two changes that produces different log files, none of them coming from this PR. The first one was regarding the changes of the impulse dynamics, and I corrected it. I did the same with the changes of the codegen in the @proyan branch. The former action was merged first. The second was done later and it has to be resolved conflict which values that doesn't reflect both changes.

I am constantly changing these values, that's why I have a perfect track of it. However we really need an unittest for this.

Btw, those values are for the Box-DDP, which doesn't solve the problem.

Reply:
I understand those are the values for the regular Box-DDP; what we see here is that it gets a lot more difficult for a regular Box-DDP to solve this problem: So it's either a disimprovement of the Box-DDP in one of the recent PRs or that the problem is much harder to solve? From following the Codegen PR, I did not see any changes that directly explain this numerical change - but I didn't read it too closely. Do we know this log file change is then due to the impulse dynamics update? Or another recent update and we forgot to look into the logs?

Reply:
In that PR, @proyan has improved the efficiency of many matrix multiplications (e.g.in the activation models and euler integrator). They produce tiny numerical differences.

Tomorrow I will tell you what happens precisely. I don't remember now what of the two changes (impulse or codegen) is the source.

Reply:
Why was this marked resolved? Did you find the source of these changes to the log?

Reply:
> Why was this marked resolved?

My mistake, I didn't notice that my message were not posted (my home internet is not that great)




>  Did you find the source of these changes to the log?

Yeap. You merged the impulse dynamics into your `codegen2` branch. It happens after my modification of the log files in this branch. Therefore, the impulse dynamics changes overwrote the ones relate to the codegen.

Unfortunately, we forgot to re-do the log file generation after that. To be honest, I was aware of it, but those days very busy. Additionally, I didn't want you to ask it, because it could produce other differences relates to your hardware.

Let say that for the moment this is my own unittest, and I am not sure if we could get robust values to be tested in the CI.

Reply:
@nim65s and I do not know why there are differences between hardware, or probably software versions. Who knows?